### PR TITLE
yaml: Switch to new Android device and kernel

### PR DESCRIPTION
**Pull request by svlad-90**
yaml: remove deprecated 'ENABLE_VIRTIO' parameter

Removed deprecated 'ENABLE_VIRTIO' parameter from the original
'prod-devel-rcar-agl.yaml'. After the introduction of the
'prod-devel-rcar-virtio.yaml' it has become redundant.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

----

yaml: Switch to the new Android device and kernel

In order to proceed with virtio functionality we need to use a dedicated
AOSP device - xenvm_trout_arm64. It was inherited from the Google Trout
device, which is intended to use virtio for device sharing. This
patch switches to the usage of that device.

xenvm_trout_arm64 device is intended to work with the 14-6.1 Android
kernel. The new kernel contains support for virtio and Xen hypervisor. So,
we are adding usage of the new kernel, as otherwise build will fail.

This AOSP device is built with additional vendor_boot and vbmeta_system
images, thus Rouge declarations were extended as well.

Signed-off-by: Vladyslav Goncharuk <vladyslav_goncharuk@epam.com>

### CODE REVIEW

File: prod-devel-rcar-virtio.yaml [hunk 1]
```diff
@@ -26,6 +26,10 @@ variables:
   XT_PREBUILT_GSX_DIR: ""
   XT_USE_DDK1_11: "0"
   XT_MACHINEOVERRIDES_RAM: "mem8gb"
+  ANDROID_KERNEL_BUILD_CONFIG: "//common-modules/xen-virtual-device:xen_virtual_device_aarch64_dist"
```
Comment (lorc) [inline prod-devel-rcar-virtio.yaml:29]: Is double slash required at the beginning of this string?

File: prod-devel-rcar-virtio.yaml [hunk 2]
```diff
@@ -44,6 +48,7 @@ common_data:
     - type: git
       url: "https://github.com/xen-troops/meta-xt-rcar.git"
       rev: master
+
```
Comment (lorc) [inline prod-devel-rcar-virtio.yaml:51]: Stray change

**Review by lorc — COMMENTED** [review]
No objections from my side.

@arminn, could you please take a look?

File: prod-devel-rcar-virtio.yaml (hunk 2 above)
Comment (svlad-90) [reply prod-devel-rcar-virtio.yaml]: Hi @lorc, removed.

File: prod-devel-rcar-virtio.yaml [hunk 3]
```diff
@@ -321,46 +325,52 @@ components:
...
+        - "--verbose_failures"
+        - "--sandbox_debug"
+      target: "%{ANDROID_KERNEL_BUILD_CONFIG}"
+      target-patterns:
+        - "--dist_dir=%{ANDROID_KERNEL_DEPLOY_DIR}"
       target_images:
-        - "out/android12-5.4/common/arch/arm64/boot/Image"
-        - "out/android12-5.4/staging/vendor/lib/modules/pvrsrvkm.ko"
+        - "%{ANDROID_KERNEL_DEPLOY_DIR}/Image"
+        - "%{ANDROID_KERNEL_DEPLOY_DIR}/initramfs.img"
```
Comment (arminn) [inline prod-devel-rcar-virtio.yaml:345]: do you really use initramfs.img from kernel build ?

Comment (svlad-90) [reply prod-devel-rcar-virtio.yaml]: Hi @arminn, this dependency once was actual. But at the current moment, we do not unpack initramfs.img. Thus, I've removed it.

**Comment by arminn** [comment]
Acked-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

**Pull request merged**